python/lsst/meas/modelfit/psf/psf.cc

```diff
@@ -0,0 +1,222 @@
+    clsAlgorithm.def_static("makeObjective", &Algorithm::makeObjective, "moments"_a, "ctrl"_a, "psfImage"_a);
+    clsAlgorithm.def_static("fitProfile", &Algorithm::fitProfile, "result"_a, "ctrl"_a, "psfImage"_a);
+    clsAlgorithm.def_static("fitShapelets", &Algorithm::fitShapelets, "result"_a, "ctrl"_a, "psfImage"_a);
+    clsAlgorithm.def("measure", &Algorithm::measure, "measRecord"_a, "exposure"_a);
+    clsAlgorithm.def("fail", &Algorithm::fail, "measRecord"_a, "error"_a=nullptr);
+}
+
+void declareGeneral(py::module & mod) {
+    using ComponentControl = GeneralPsfFitterComponentControl;
+    using Control = GeneralPsfFitterControl;
+    using Fitter = GeneralPsfFitter;
+    using Algorithm = GeneralPsfFitterAlgorithm;
+
+    using PyComponentControl = py::class_<ComponentControl, std::shared_ptr<ComponentControl>>;
+    using PyControl = py::class_<Control, std::shared_ptr<Control>>;
+    using PyFitter = py::class_<Fitter, std::shared_ptr<Fitter>>;
+    using PyAlgorithm = py::class_<Algorithm, std::shared_ptr<Algorithm>, Fitter>;
+
+    PyComponentControl clsComponentControl(mod, "GeneralPsfFitterComponentControl");
+    clsComponentControl.def(py::init<int, double>(), "order"_a=0, "radius"_a=1.0);
+    LSST_DECLARE_CONTROL_FIELD(clsComponentControl, ComponentControl, order);
+    LSST_DECLARE_CONTROL_FIELD(clsComponentControl, ComponentControl, positionPriorSigma);
+    LSST_DECLARE_CONTROL_FIELD(clsComponentControl, ComponentControl, ellipticityPriorSigma);
+    LSST_DECLARE_CONTROL_FIELD(clsComponentControl, ComponentControl, radiusFactor);
+    LSST_DECLARE_CONTROL_FIELD(clsComponentControl, ComponentControl, radiusPriorSigma);
+
+    PyControl clsControl(mod, "GeneralPsfFitterControl");
+    clsControl.def(py::init<>());
+    LSST_DECLARE_NESTED_CONTROL_FIELD(clsControl, Control, inner);
+    LSST_DECLARE_NESTED_CONTROL_FIELD(clsControl, Control, primary);
+    LSST_DECLARE_NESTED_CONTROL_FIELD(clsControl, Control, wings);
+    LSST_DECLARE_NESTED_CONTROL_FIELD(clsControl, Control, outer);
+    LSST_DECLARE_NESTED_CONTROL_FIELD(clsControl, Control, optimizer);
+    LSST_DECLARE_CONTROL_FIELD(clsControl, Control, defaultNoiseSigma);
+
+    PyFitter clsFitter(mod, "GeneralPsfFitter");
+    clsFitter.def(py::init<Control const &>(), "ctrl"_a);
+    clsFitter.def("addFields", &Fitter::addFields, "schema"_a, "prefix"_a);
+    clsFitter.def("getModel", &Fitter::getModel);
+    clsFitter.def("getPrior", &Fitter::getPrior);
+    clsFitter.def("adapt", &Fitter::adapt, "previousFit"_a, "previousModel"_a);
+    clsFitter.def(
+        "apply",
+        [](
+            Fitter const & self,
+            afw::image::Image<Pixel> const & image,
+            afw::geom::ellipses::Quadrupole const & moments,
+            Scalar noiseSigma
+        ) {
+            return self.apply(image, moments, noiseSigma);
+        },
+        "image"_a,
+        "moments"_a,
+        "noiseSigma"_a=-1
+    );
```

Comment:
The original signature has an optional argument (`int*` used as an optional output) not relevant for Python.  Comment added to that effect.